app/models/user.rb

```diff
@@ -302,7 +302,7 @@ def locked?
   end
 
   def self.search(term)
-    term.present? ? where("email = ? OR username ILIKE ?", term, "%#{term}%") : none
+    term.present? ? where("email = ? OR username ILIKE ?", term.strip, "%#{term.strip}%") : none
```

Comment:
I know it is not a big deal but I was wondering how to call strip only once. This is the only way a managed to do it:
```
  def self.search(term)
    return none if term.blank?

    term.strip!
    where("email = ? OR username ILIKE ?", term, "%#{term}%")
  end
```

Reply:
I'm not fond of modifying parameters inside a method call :thinking:. Maybe in this case it's OK, but the scenario could be error-prone in some situations. Consider this console session:

```ruby
term = " abc "
=> " abc "
User.search(term)
term
=> "abc"
```

Very unintuitive IMHO.

Reply:
Yes, you are totally right @javierm ☺️ ! Thanks

So, what about this version?
```
  def self.search(term)
    return none if term.blank?

    search = term.strip
    where("email = ? OR username ILIKE ?", search, "%#{search}%")
  end
```

Maybe to keep the duplication is the best option here keeping the one line implementation. 🤔

Reply:
No strong preference.